src/Tickets/Commerce/Order_Modifiers/Modifiers/Coupon.php

```diff
@@ -48,14 +48,16 @@ class Coupon extends Modifier_Abstract {
 	];
 
 	/**
-	 * Constructor for the Coupon strategy.
+	 * Retrieves the display name of the modifier in singular or plural form.
 	 *
 	 * @since TBD
+	 *
+	 * @param bool $plural Whether to return the plural form. Defaults to false (singular).
+	 *
+	 * @return string The display name of the modifier.
 	 */
-	public function __construct() {
-		parent::__construct( $this->modifier_type );
-		$this->modifier_display_name        = __( 'Coupon', 'event-tickets' );
-		$this->modifier_display_name_plural = __( 'Coupons', 'event-tickets' );
+	public function get_modifier_display_name( bool $plural = false ): string {
+		return _n( 'Coupon', 'Coupons', ( (int) $plural ) + 1, 'event-tickets' );
```

Comment:
Why not a separate method for the plural form? This seems like a hacky way to get the plural form, since `_n()` is meant for a dynamic plural vs. a pre-known plural.

Reply:
I think, since the method `get_modifier_display_name` is already there and accepts a $plural param, i would use it.

i would also like to avoid code in other places if possible. This is already on QA. Actually has passed QA :( 

I agree that the casting from bool to int + 1 is a bit hacky but produced the desired.

Reply:
we do actually have separate functions for ticket vs tickets, for example.
+1